src/network.cpp

```diff
@@ -64,13 +66,59 @@ namespace oxen::quic
 
     void Network::close()
     {
-        quic_manager->close_all();
-        std::this_thread::sleep_for(std::chrono::milliseconds(2500));
-    }
+        if (not running.exchange(false))
+            return;
 
-    void Network::run()
-    {
-        ev_loop->run();
+        log::info(log_cat, "Shutting down context...");
+
+        std::promise<void> p;
+        auto f = p.get_future();
+        quic_manager->call([this, &p]() {
+            try
+            {
+                quic_manager->close_all();
+
+                for (auto& [addr, handle] : mapped_client_addrs)
+                {
+                    auto s = handle->sock();
+                    log::debug(log_cat, "closing client UDPHandle bound to {}:{}", s.ip, s.port);
+                    handle->close();
+                }
+                for (auto& [addr, handle] : mapped_server_addrs)
+                {
+                    auto s = handle->sock();
+                    log::debug(log_cat, "closing server UDPHandle bound to {}:{}", s.ip, s.port);
+                    handle->close();
+                }
```

Comment:
Nit-pick: I know that if `mapped_{client,server}_addrs` is empty, it will just exit out of the for-loop without iterating through it, but for some reason not explicitly checking sticks out to me in a bad way

Nittier-pick: it could probably be `const auto&`, since calling `handle->close()` doesn't modify the underlying object

Reply:
const-ified, mildly begrudgingly